### PR TITLE
Update git.rst

### PR DESCRIPTION
**Pull request by tschmetzer**
I added welcoming message and a repo fork description for contributors

### CODE REVIEW

**Comment by tschmetzer** [comment]
Anybody to review my text to facilitate winning more developers? :)

**Review by DelazJ — COMMENTED** [review]
Hit @tschmetzer
Sorry for the very late feedback. Not really supportive/welcoming.
I have some concerns with parts of this PR (I may be biased). We have written a step by step guidelines to contribute to docs, a great part of this is mainly about GitHub. I'd assume we would rely on/adapt that if we need to show people how to proceed with GH. One advantage of our instructions is that there's a profile split, between local setup and online modification. People imho do not necessarily need a local clone to contribute (even to the code repo) so we shouldn't push them to do it unless they need it (due to the complexity of their changes)

**Comment by tschmetzer** [comment]
> 
> 
> Hit @tschmetzer
> Sorry for the very late feedback. Not really supportive/welcoming.
> I have some concerns with parts of this PR (I may be biased). We have written a step by step guidelines to contribute to docs, a great part of this is mainly about GitHub. I'd assume we would rely on/adapt that if we need to show people how to proceed with GH. One advantage of our instructions is that there's a profile split, between local setup and online modification. People imho do not necessarily need a local clone to contribute (even to the code repo) so we shouldn't push them to do it unless they need it (due to the complexity of their changes)

Hi @DelazJ 
no worries, thanks for replying!
I consider it very valuable that there are already guidelines for doc contributions with GH (I guess you are referring to https://docs.qgis.org/3.16/en/docs/documentation_guidelines/first_contribution.html#id10)
I agree that especially for documentation contributions in most cases a local setup is not required. 

Yet when looking at the developers guide where the cloning a repository for contribution is described (I mean this git.rst) I would expect all required steps to be shown in the proceeding. Because as far as I know a contributor is not able to work on master and I guess this documentation is not meant for proficient collaborators but for new contributors.

If you prefer to have only one description/source for both the documenation guide and the developers guide maybe it would be a an idea to think about a unique .rst file that can be included into both guides. I don't know if something similar to a C language #include statement exists in the .rst world.

For me it would be enough to add the changes we're already discussing.

File: docs/developers_guide/git.rst [hunk 1]
```diff
@@ -155,6 +158,24 @@ first and merged to master (the main branch) when they are finished and stable.
 This section describes the procedure for branching and merging in the QGIS
 project.
 
+Roles
+-----
+
+There exist various roles on GitHub. When having an account on GitHub you are already allowed to contribute by forking the repository and have the role 'contributor'. Core developers are 'collaborators' and can merge branches into the master repository.
```
[the remark this answers is not in the document]
Comment (tschmetzer) [reply docs/developers_guide/git.rst:164]: > 
> 
> master is not a repo, and they can merge in any branches.
> Also, where does the 'collaborator' name come from? U sure?

I am absolutely not sure, I my source is this proceeding from a different issue: 
grafik

What I am sure about is that there seem to exist different roles. This naming and their responsibilities should briefly be described. It should be clear that usual contributors are unable to merge and it should be clear what role actually can merge branches.

@DelazJ : Who is able to make a remark or confirm regarding the roles?

File: docs/developers_guide/git.rst [hunk 2]
```diff
@@ -155,6 +158,24 @@ first and merged to master (the main branch) when they are finished and stable.
...
+
+Contributions to development
+----------------------------
+
+Once signed up on GitHub you can engage as a contributor. You can contribute by forking the repository and initiating a pull request. To do that follow these generic steps:
+
+1. Fork the repository on GitHub
+2. Clone it onto your local computer and set up the build environment
```
[the remark this answers is not in the document]
Comment (tschmetzer) [reply docs/developers_guide/git.rst]: > 
> 
> Is this necessary, particularly having to set up a build? One can contribute to QGIS code without a local clone, and build. Many beginners changes can be done in github website imho. Unless the changes are quite intrusive...

I added this info in a different commit

> 
> And if I need to build, how/where are instructions?

Aren't the instructions to build in the next chapter https://docs.qgis.org/3.16/en/docs/developers_guide/qtcreator.html?

I was pointed to KartozaGeo's youtube video channel for the setup where he runs through
setting up a development environment: https://www.youtube.com/c/KartozaGeo/videos
But his video essentially reflects the instructions in the next chapter.

**Review by DelazJ — COMMENTED** [review]
> Yet when looking at the developers guide where the cloning a repository for contribution is described (I mean this git.rst) I would expect all required steps to be shown in the proceeding. Because as far as I know a contributor is not able to work on master and I guess this documentation is not meant for proficient collaborators but for new contributors.

You are right, indeed. I have to revise my comments. Looks like I did not well read this file, and now that I do, it looks to me that the issue does not reside in your pull-request (you are following the logic in this file) but to how this chapter itself is organized and the assumptions it's based on. There are few things worth an update imho.
\<pedantic mode>: everyone can work on master. master is simply a branch you have in your fork repo so you can work on it if you want. By default, it is(/was?) the default branch so there's a convention(!) that you better not alter it and just use it to track the master branch in the upstream repo. Which is a fine advice for beginners, though in GH they have no easy way to do this tracking. The place where most of us can NOT directly work is within the qgis/QGIS repository branches.</>

> Who is able to make a remark or confirm regarding the roles?

GitHub docs, probably. In my context, I see "contributor" and "member", but probably because I'm myself a QGIS project member ? So, let's go with what you have.

Few typo, split to commit and I'll merge.

**Pull request merged**